wimm.Guardian.UnitTests/EnumArgumentExtensionsTests.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace wimm.Guardian.UnitTests
+{
+    [TestClass]
+    public class EnumArgumentExtensionsTests
+    {
+        [TestMethod]
+        public void IsDefined_ValueUndefined_Throws()
+        {
+            var underTest = new Argument<TestEnum>("Doot", (TestEnum)(int.MaxValue));
+            Assert.ThrowsException<EnumArgumentOutOfRangeException>(() => underTest.IsDefined());
+        }
+
+        [TestMethod]
+        public void IsDefined_ValueNotEnum_Throws()
+        {
+            var underTest = new Argument<DateTime>("Doot", DateTime.Now);
+            Assert.ThrowsException<TypeArgumentException>(() => underTest.IsDefined());
```

Comment:
Not that I have a particular expectation of wanting to use `IsDefined` on another type, but it seems kind of yucky that the calling code gives no hints of why this would fail. What do you think of `IsDefinedEnum` instead?

Reply:
I think that's reasonable. 👍 wf